app/services/ProductServiceImpl.java

```diff
@@ -1,43 +1,40 @@
 package services;
 
-import exceptions.ProductNotFoundException;
-import io.sphere.sdk.client.SphereClient;
+import com.google.inject.Singleton;
+import io.sphere.sdk.client.PlayJavaSphereClient;
 import io.sphere.sdk.products.ProductProjection;
-import io.sphere.sdk.products.ProductVariant;
 import io.sphere.sdk.products.queries.ProductProjectionQuery;
 import io.sphere.sdk.queries.PagedQueryResult;
+import play.Logger;
+import play.inject.ApplicationLifecycle;
+import play.libs.F;
 
+import javax.inject.Inject;
 import java.util.Optional;
-import java.util.concurrent.CompletionStage;
-
-import static java.util.Objects.requireNonNull;
 
+@Singleton
 public class ProductServiceImpl extends AbstractShopService implements ProductService {
 
-    private final ProductProjection cachedProduct;
+    private static final Logger.ALogger LOG = Logger.of(ProductServiceImpl.class);
 
-    public ProductServiceImpl(final SphereClient sphereClient) {
-        super(sphereClient);
-        this.cachedProduct = loadProduct().orElseThrow(ProductNotFoundException::new);
-    }
+    private final F.Promise<Optional<ProductProjection>> cachedProduct;
 
-    @Override
-    public Optional<ProductVariant> getVariantFromId(ProductProjection product, int variantId) {
-        requireNonNull(product, "'product' must not be null");
-        return Optional.ofNullable(product.getVariant(variantId));
+    @Inject
+    public ProductServiceImpl(final ApplicationLifecycle applicationLifecycle,
+                              final PlayJavaSphereClient playJavaSphereClient) {
+        super(applicationLifecycle, playJavaSphereClient);
+        this.cachedProduct = loadProduct();
     }
 
-    private Optional<ProductProjection> loadProduct() {
+    private F.Promise<Optional<ProductProjection>> loadProduct() {
```

Comment:
Now that we use DI, wouldn't it be better to obtain this product in the `DonutShopModule` and get rid of this class?